src/storage/src/storage/read_object.rs

```diff
@@ -1310,21 +1312,41 @@ mod tests {
         Ok(())
     }
 
-    #[test_case(0, -100, RangeError::NegativeLimit; "negative limit")]
-    #[test_case(-100, 100, RangeError::NegativeOffsetWithLimit; "negative offset with positive limit")]
     #[tokio::test]
-    async fn test_range_header_error(offset: i64, limit: i64, want_err: RangeError) -> Result {
+    async fn range_header_negative_limit() -> Result {
         let inner = test_inner_client(test_builder());
         let err = ReadObject::new(inner, "projects/_/buckets/bucket", "object")
-            .with_read_offset(offset)
-            .with_read_limit(limit)
+            .with_read_limit(-100)
```

Comment:
Should this type be a `u64`?

Or is that too annoying bc we need the offset to accept negative values

Reply:
See #2849, I think we need to change the input type to something that makes these errors impossible.

Reply:
Ah, yeah, I like your proposal